dissect/target/plugins/apps/productivity/msoffice.py

```diff
@@ -0,0 +1,413 @@
+                )
+
+    def _lookup_com_executable(self, prog_id: str) -> str | None:
+        """Lookup the com executable given a prog id using the registry."""
+
+        for classes_root in self.CLASSES_ROOTS:
+            try:
+                cls_id = self.target.registry.value(f"{classes_root}\\{prog_id}\\CLSID", "(Default)").value
+                inproc_key = f"{classes_root}\\CLSID\\{cls_id}\\InprocServer32"
+                return self.target.registry.value(inproc_key, "(Default)").value
+            except RegistryError:
+                pass
+
+    def _parse_vsto_manifest(self, manifest_path_str: str, user_sid: str) -> set[str]:
+        """Parse a vsto manifest.
+
+        Non-local manifests, i.e. not ending with suffix "vstolocal" are listed but skipped.
+        """
+
+        if not manifest_path_str.endswith("vstolocal"):
+            self.target.log.warning("Parsing of remote vsto manifest %s is not supported")
+            return set(manifest_path_str)
+
+        manifest_path: Path = self.target.resolve(manifest_path_str.removesuffix("|vstolocal"), user_sid)
+        manifest_parser = ClickOnceDeploymentManifestParser(manifest_path, self.target, user_sid)
+        return manifest_parser.find_codebases(manifest_path)
+
+    def _parse_web_addin_manifest(self, manifest_path: Path) -> OfficeWebAddinRecord:
+        """Parses a web addin manifest."""
+
+        ns = {"": "http://schemas.microsoft.com/office/appforoffice/1.1"}
+
+        manifest_tree: Element = ElementTree.fromstring(manifest_path.read_text("utf-8-sig"))
+
+        source_location_elements = manifest_tree.findall(".//SourceLocation", ns)
+        source_locations = [source_location.get("DefaultValue") for source_location in source_location_elements]
+
+        display_name_element = manifest_tree.find(".//DisplayName", ns)
+        display_name = display_name_element.get("DefaultValue") if display_name_element is not None else None
+
+        return OfficeWebAddinRecord(
+            name=display_name,
+            manifest=manifest_path,
+            version=manifest_tree.findtext(".//Version", namespaces=ns),
+            provider_name=manifest_tree.findtext(".//ProviderName", namespaces=ns),
+            source_locations=filter(None, source_locations),
+            modification_time=manifest_path.stat().st_mtime,
+        )
+
+    def _office_install_root(self, component: Literal["Word", "Excel"]) -> str:
+        """Return the installation root for a office component"""
+
+        # Typically, all components share the same root.
+        # Curiously enough, the "Common" component has no InstallRoot defined.
+        key = f"HKLM\\{self.OFFICE_KEY}\\16.0\\{component}\\InstallRoot"
+        try:
+            return self.target.registry.value(key, "Path").value
+        except RegistryError:
+            return self.OFFICE_DEFAULT_ROOT
+
+    @lru_cache(16)
```

Comment:
Can you initialise this in the `__init__`? Search for other uses of `lru_cache` in the Dissect project for examples.

If you want context, hit me up offline.

Reply:
Posted this in an earlier comment. Some more context can also be found here https://github.com/fox-it/dissect.target/pull/547!